TNutils.py

```diff
@@ -423,8 +446,9 @@ def learning_step(mps, index, imgs, lr, going_right = True):
     # the data-dependent terms
     psifrac = 0
     for img in imgs:
-        num = computepsiprime(mps,img,index)  # PSI(v)
-        den = computepsi(mps,img)             # PSI(v')
+        num = computepsiprime(mps,img,index)    # PSI'(v)
+        # TODO: 'ijkl,ilkj' or 'ijkl,ijkl'?
```

Comment:
```suggestion
        # 'ijkl,ilkj' or 'ijkl,ijkl'?
        # computepsiprime was coded so that the ordering of the indexes is the same
        # as the contraction A = mps.tensors[index] @ mps.tensors[index+1]
        # so it should be the second one    
```